checker.cpp

```diff
@@ -1,22 +1,55 @@
-#include <assert.h>
 #include <iostream>
+#include "checker.h"
+#ifdef UTEST
+#include "checker_test.h"
+#endif //UTEST
+
 using namespace std;
 
-bool batteryIsOk(float temperature, float soc, float chargeRate) {
-  if(temperature < 0 || temperature > 45) {
-    cout << "Temperature out of range!\n";
-    return false;
-  } else if(soc < 20 || soc > 80) {
-    cout << "State of Charge out of range!\n";
+bool InRange(float min, float max, float value){
```

Comment:
Nice abstraction and overloading.
One thought: if we needed to change the `> max` to `>= max`, we would need to do it in two places.
Can you think of an alternate way where we would need to do that in one place only?

Reply:
I think I have a new proposal for this algorithm thanks for the hint!